core/shared/src/main/scala/scalaz/zio/Void.scala

```diff
@@ -0,0 +1,8 @@
+package scalaz.zio
+
+/**
+ * Represents something that can't exist
```

Comment:
I'd say:

```
An uninhabited type. Because there are no values of this type, it represents a guarantee, at compile-time, that values of type `Void` may not exist. The `absurd` method can be used to eliminate terms of a sum type that have type `Void`, e.g. `Either[Void, A]` may be simplified via `absurd` to `A`.

This type is similar to `Nothing` but does not suffer from weird edge cases in Scalac.
```

Reply:
Done. I've also referenced https://github.com/scala/bug/issues/9453 to give the reader better clue about problems with `Nothing`.